lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1453,12 +1464,16 @@ instance FromHttpApiData ApiPoolId where
         | t == "*" =
             Right ApiPoolIdPlaceholder
         | otherwise =
-            bimap pretty ApiPoolId (fromText t)
+            ApiPoolId <$> case fromText t of
+                Left _ ->
+                    left (T.pack . displayException) $ decodePoolIdBech32 t
```

Comment:
I'd be best to specialize to `Either TextDecodingError` here and use `ShowFmt` as for other error renderer here. Otherwise, I am afraid that the `displayException` will default to `Show` and renders the error tainted with Haskell syntax instead of something more human-readable for the API.

Reply:
In this case, I wrote a non-Show instance for Exception though